lib/services/libvirtd.pm

```diff
@@ -115,10 +134,17 @@ sub full_libvirtd_check {
         shutdown_guest(%guest_params) if get_var("NESTED_VM_DOWN");
     }
 
-    common_service_action($pkg, $type, 'is-enabled');
-    common_service_action($pkg, $type, 'is-active');
+    if (!(($stage eq 'after') && (get_var("VERSION") eq "Tumbleweed"))) {
+	common_service_action($pkg, $type, 'is-enabled');
+	common_service_action($pkg, $type, 'is-active');
+    }
 
     if ($stage eq 'after') {
+	if (get_var("VERSION") eq "Tumbleweed") {
+	    enable_upgraded_service();
```

Comment:
Others looks good to me, just feel a bit strange why need 3 functions here and each one only invoke once, could we just reserve one function here?

Reply:
Updated it to one function.